arbnode/transaction_streamer.go

```diff
@@ -522,6 +532,41 @@ func (s *TransactionStreamer) GetMessage(seqNum arbutil.MessageIndex) (*arbostyp
 	return &message, nil
 }
 
+func (s *TransactionStreamer) getMessageWithMetadataAndBlockHash(seqNum arbutil.MessageIndex) (*arbostypes.MessageWithMetadataAndBlockHash, error) {
+	msg, err := s.GetMessage(seqNum)
+	if err != nil {
+		return nil, err
+	}
+
+	// Get block hash.
+	// First check if key exists in database so this procedure is backwards compatible
+	// with databases' snapshots that don't have block hashes stored.
+	key := dbKey(blockHashInputFeedPrefix, uint64(seqNum))
+	hasBlockHash, err := s.db.Has(key)
```

Comment:
I tried to use the same strategy deployed [here](https://github.com/OffchainLabs/nitro/blob/1b74fae1c4548249cc68bf0461b7cff6e01265ab/arbnode/transaction_streamer.go#L651), to avoid using `Has` checking if `Get` returns an error that is leveldb.ErrNotFound or pebble.ErrorNotFound, but it didn't properly work with system tests.